src/NUnitFramework/framework/Constraints/StringConstraint.cs

```diff
@@ -73,16 +93,44 @@ protected StringConstraint(string expected)
 
         /// <summary>
         /// Modify the constraint to ignore case in matching.
+        /// This will call Using(StringComparison.CurrentCultureIgnoreCase).
         /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when a comparison type different
+        /// than <see cref="StringComparison.CurrentCultureIgnoreCase"/> was already set.</exception>
         public virtual StringConstraint IgnoreCase
         {
             get
             {
+                Using(StringComparison.CurrentCultureIgnoreCase);
                 caseInsensitive = true;
                 return this;
             }
         }
 
+        /// <summary>
+        /// Modify the constraint to use the specified comparison.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when a comparison type different
+        /// than <paramref name="strComparison"/> was already set.</exception>
+        public virtual StringConstraint Using(StringComparison strComparison)
```

Comment:
I dislike shorthand, please change `strComparison` to `comparisonType`.
Mainly to match the one that was in `SubstringContraint`.
Once you renamed the field to `_comparisonType` you won't have conflicts.